Color by sampling year

kimandrews opened the pull request:
## Description of proposed changes
Since we're not inferring a timetree, this PR allows visualization of the sampling date by:
* Enabling coloring by sampling year
* Allowing sampling date to appear when clicking on tips and as a filtering option

Example output: https://nextstrain.org/staging/tb/colorbyyear2?c=year

Coloring by year is accomplished through the following steps:
* Extract year from each sample's date field
* Output a JSON file with node data containing year for each sample
* Generate a TSV file that assigns colors to each year
* Concatenate the new TSV file with the defaults/colors.tsv file
* Export year and color metadata to auspice

The first 3 steps above are accomplished using a [script copied from ebola](https://github.com/nextstrain/ebola/blob/9aa7403a9aac47133f29fdb6850647377067f312/phylogenetic/scripts/get_year.py), with these modifications:
* Generate a TSV file that assigns colors to each year (previously colors were only suggested)
* Allow coloring for up to 50 unique years (previously only 36 unique years were allowed)

## Related issue(s)
This addresses part of https://github.com/nextstrain/tb/issues/34, but does not address recency coloring

## Checklist

<!--
Make sure checks are successful at the bottom of the PR.

If applicable, add:
- any changes to existing tests
- any additional manual testing to confirm changes

Please add a note if you need help with adding tests.
-->

- [ ] Checks pass
- [ ] Update changelog

<!-- 🙌 Thank you for contributing to Nextstrain! ✨ -->

jameshadfield reviewed (approved):
Code looks good (but didn't test)

If you change the (auspice-config) coloring to be categorical you can display the legend in different ways, but it's a little more work as you would want to define the ordering of the years so that they're chronological. This is what I did for [ebola](https://next.nextstrain.org/ebola/all-outbreaks)

<img width="226" height="232" alt="Image" src="https://github.com/user-attachments/assets/1930c3a3-242f-4046-b4ad-5ab654c6e203" />

jameshadfield commented:
> I think it should be straight-forward to copy the [get-year.py script from ebola ](https://github.com/nextstrain/ebola/blob/01e373e67b865af5a8d0f28daf26196375e36f3b/phylogenetic/scripts/get_year.py)and use it here.

One improvement (which I have been meaning to make there): rather than suggest a colouring block for the JSON, write them out in a `colors.tsv` which (I think!) will still result in the desired legend ordering.

kimandrews committed:
Copy get_year.py from ebola

Copied this script: https://github.com/nextstrain/ebola/blob/9aa7403a9aac47133f29fdb6850647377067f312/phylogenetic/scripts/get_year.py

kimandrews committed:
Add year-based coloring

  - Change get_year.py to write colors TSV
  - Add get_year and combine_colors rules to generate year colors
  - Update export rule to include year node data
  - Configure year coloring in auspice config

kimandrews committed:
Extend color scheme

Extend color schemes to support up to 50 years, since minimum sampling date is 1990.

joverlee521 reviewed (approved):
Changes look good to me 🌈
